File: runtime/metadata/metadata.go
// Copyright 2026 dywoq - Apache License 2.0
// Part of Vacui SDK: https://github.com/dywoq/vacui

// Package metadata contains declarations of the game metadata.
package metadata

type Version struct {
	View      string `json:"view"`
	Changelog string `json:"changelog"`
}

type Author struct {
	Name string `json:"name"`
}

type Metadata struct {
	Title    string    `json:"title"`
	Authors  []Author  `json:"authors"`
	Versions []Version `json:"versions"`
	License  string    `json:"license"`
	Links    []string  `json:"links"`
}

File: krnl/pal/qemu/metadata.cxx
// Copyright 2026 dywoq - Apache License 2.0
// A part of https://github.com/dywoq/vacui
//
// Description:
//
// QEMU platform implementation of vacui/pal/metadata.h

#include <vacui/pal/metadata.h>

namespace vacui::pal {
    const char *get_name() noexcept { return "qemu"; }
    id          get_id() noexcept { return id::qemu; }
} // namespace vacui::pal
